File: src/Core/Scripts/Runtime/Array.js
// Array Extensions

extend(Array.prototype, {
  clear: function() {
    this.length = 0;
  },

  contains: function(item) {
    return (this.indexOf(item) >= 0);
  },

  getRange: function(index, count) {
    return this.slice(index, index + count);
  },

  insert: function(index, item) {
    this.splice(index, 0, item);
  },

  insertRange: function(index, items) {
    if (index === 0) {
      this.unshift.apply(this, items);
    }
    else {
      for (var i = 0; i < items.length; i++) {
        this.splice(index + i, 0, items[i]);
      }
    }
  },

  remove: function(item) {
    var index = this.indexOf(item);
    if (index >= 0) {
      this.splice(index, 1);
      return true;
    }
    return false;
  },

  removeAt: function(index) {
    this.splice(index, 1);
  },

  removeRange: function(index, count) {
    return this.splice(index, count);
  }
});

File: src/DSharp.Mscorlib/Array.cs
using System.Collections;
using System.Collections.Generic;
using System.Runtime.CompilerServices;

namespace System
{
    // NOTE: Keep in sync with ArrayList and List
    [ScriptIgnoreNamespace]
    [ScriptImport]
    [ScriptName("Array")]
    public sealed partial class Array : IList, ICollection, IEnumerable
    {
        [ScriptField]
        public extern object this[int index]
        {
            get;
            set;
        }

        [ScriptField]
        [ScriptName("length")]
        public extern int Length { get; }

        public extern int Count { get; }

        public extern int Add(object value);

        public extern void Clear();

        public extern bool Contains(object value);

        public extern IEnumerator GetEnumerator();

        public extern int IndexOf(object value);

        public extern void Insert(int index, object value);

        public extern void Remove(object value);

        public extern void RemoveAt(int index);

        public extern static explicit operator List<object>(Array array);
    }
}
